[IMP] inventory: update locations doc (#8906)
odoo/documentation: content/applications/inventory_and_mrp/inventory/warehouses_storage/inventory_management/use_locations.rst

Locations
=========

> In Odoo *Inventory*, a *location* is a specific space within a warehouse. This can be a shelf, room,
> aisle, rack, bin, etc.

Reviewer: There are also virtual and external locations, too! If you'd like to focus on the internal locations, I'd rewrite this to something like:
```suggestion
In Odoo *Inventory*, a *location* is a specific space within the supply chain. Locations within the warehouse (e.g. shelf, room,
aisle, rack, bin, etc) can be used to .... [your next sentence]
```

Author: Added a sentence about virtual and external locations. Suggestion implemented ✅

Locations allow for the storage of products in designated areas of the warehouse, to help track
inventory accurately.

Configuration
=============

> In order to use locations, the :guilabel:`Storage Locations` setting must be enabled. To do so,

Reviewer: Optional add for specificity (:
```suggestion
To specify storage locations, the :guilabel:`Storage Locations` setting must be enabled. To do so,
```

Author: Suggestion implemented ✅

open a list of all existing locations in the warehouse.

Create a new location inside a warehouse
========================================

To create a new location inside a warehouse, navigate to :menuselection:`Inventory app -->
Configuration --> Locations`. Doing so opens a list of all existing locations. Click :guilabel:`New`
to open a new location creation form.

In the :guilabel:`Location Name` field, assign a name for the new location. In the :guilabel:`Parent
Location` field, assign the location that this new location exists in.

.. example::
   A new location `Gate 1` is created. Its parent location is `Dispatch Zone`. In the
   :guilabel:`Parent Location` for `Gate 1`, select `Dispatch Zone`.

In the :guilabel:`Additional Information` section, fill out the necessary details for the new
location.

In the :guilabel:`Location Type` field, select which type of location this is. There are seven
different types of locations to choose from:

- :guilabel:`Vendor Location`: Virtual location representing the source location for products coming
  from vendors.
- :guilabel:`View`: Virtual location used to create a hierarchical structure for a warehouse;

>   aggregates its child locations. **Cannot directly contain products**.

Reviewer: > Cannot directly contain products

It can, though it shouldn't, so let's reword that (:

```suggestion
  aggregates its child locations. **Should not directly contain products**.
```

Author: Suggestion implemented ✅

- :guilabel:`Customer Location`: Virtual location representing the destination location for products
  sent to customers.
- :guilabel:`Inventory Loss`: Virtual location serving as counterpart for inventory operations. Used
  to correct stock levels of physical inventory.
- :guilabel:`Production`: Virtual counterpart location for production operations. This location
  consumes components and produces finished products.
- :guilabel:`Transit Location`: Counterpart location that should be used in inter-company or
  inter-warehouse operations.

.. image:: use_locations/use-locations-new-location-form.png
   :align: center
   :alt: Additional Information section of new location creation form.

Configure the remaining fields in the :guilabel:`Additional Information` field as follows:

- :guilabel:`Company`: the company whose warehouse the location is inside of. Leave this field blank
  if this location is shared between companies.
- :guilabel:`Is a Scrap Location?`: check this box to allow for scrapped/damaged goods to be stored
  in this location.
- :guilabel:`Is a Return Location?`: check this box to allow products to be returned to this
  location.

> - :guilabel:`Barcode`: the barcode number assigned to the location.

Reviewer: location barcodes can also be letters (:
```suggestion
- :guilabel:`Barcode`: the barcode assigned to the location.
```

Author: Nice catch - Suggestion implemented ✅

.. image:: use_locations/use-locations-cyclic-counting.png
   :align: center
   :alt: Cyclic Counting section of new location creation form.

When different than `0`, inventory count date for products stored at this location are automatically
set at the defined frequency.

.. _inventory/location-hierarchy:

Create location hierarchies
===========================

The *Parent Location* setting on the new location form allows for a location to exist within a
warehouse, or below another location.

Every location can serve as a parent location, and every parent location can have multiple locations
within it, allowing for the creation of a hierarchical structure.

.. example::
   Location hierarchy could be organized so that a small refrigerator is located on a specific
   shelf, which is located in the warehouse.

   .. image:: use_locations/use-locations-hierarchy.png
     :align: center
     :alt: Location hierarchy of small refrigerator, on a shelf in a warehouse.

   To create the location hierarchy in the example above, set the warehouse stock as the parent of
   the shelf, and the shelf as the parent of the small refrigerator. This can be adapted to a

>    hierarchy of any magnitude.

Reviewer: I'd also spell out the resulting location hierarchy `WH/Stock/Shelf 2/Small Refrigerator` here, as well!

Author: Suggestion implemented ✅

Locations
=========

In Odoo *Inventory*, a *location* is a specific space within a warehouse. This can be a shelf, room,
aisle, rack, bin, etc. There are also external and virtual locations, such as customer locations,

> and locations specifically designated as "virtual", instead of real, physical locations.

Author: Suggestion implemented ✅

Locations allow for the storage of products in designated areas of the warehouse, to help track
inventory accurately.

Configuration
=============

> To specify storage locations, the :guilabel:`Storage Locations` setting must be enabled. To do so,

Author: Suggestion implemented ✅

to open a new location creation form.

In the :guilabel:`Location Name` field, assign a name for the new location. In the :guilabel:`Parent
Location` field, assign the location that this new location exists in.

.. example::
   A new location `Gate 1` is created. Its parent location is `Dispatch Zone`. In the
   :guilabel:`Parent Location` for `Gate 1`, select `Dispatch Zone`.

In the :guilabel:`Additional Information` section, fill out the necessary details for the new
location.

In the :guilabel:`Location Type` field, select which type of location this is. There are seven
different types of locations to choose from:

- :guilabel:`Vendor Location`: Virtual location representing the source location for products coming
  from vendors.
- :guilabel:`View`: Virtual location used to create a hierarchical structure for a warehouse;
  aggregates its child locations. **Should not directly contain products**.
- :guilabel:`Internal Location`: Physical location inside a warehouse.
- :guilabel:`Customer Location`: Virtual location representing the destination location for products
  sent to customers.

> - :guilabel:`Inventory Loss`: Virtual location serving as counterpart for inventory operations. Used

Author: Suggestion implemented ✅

  to correct stock levels of physical inventory.
- :guilabel:`Production`: Virtual counterpart location for production operations. This location
  consumes components and produces finished products.
- :guilabel:`Transit Location`: Counterpart location that should be used in inter-company or
  inter-warehouse operations.

.. image:: use_locations/use-locations-new-location-form.png
   :align: center
   :alt: Additional Information section of new location creation form.

> Configure the remaining fields in the :guilabel:`Additional Information` field as follows:

Author: Suggestion implemented ✅

- :guilabel:`Company`: the company whose warehouse the location is inside of. Leave this field blank
  if this location is shared between companies.

> - :guilabel:`Is a Scrap Location?`: check this box to allow for scrapped/damaged goods to be stored

Author: Suggestion implemented ✅

  in this location.

> - :guilabel:`Is a Return Location?`: check this box to allow products to be returned to this

Author: Suggestion implemented ✅

  location.
- :guilabel:`Barcode`: the barcode assigned to the location.

> - :guilabel:`Replenish Location`: check this box to get all quantities to replenish at this

Author: Suggestion implemented ✅

  location.

In the :guilabel:`Logistics` section, in the :guilabel:`Removal Strategy` field, click the drop-down
menu and select the :doc:`removal strategy <../removal_strategies>` for how items should be removed
from this location.

In the :guilabel:`Cyclic Counting` section, change the value in the :guilabel:`Inventory Frequency

> (Days)` field from the default `0`, if necessary.

Author: That's a great catch - Suggestion implemented ✅

.. image:: use_locations/use-locations-cyclic-counting.png
   :align: center
   :alt: Cyclic Counting section of new location creation form.

> When different than `0`, inventory count date for products stored at this location are automatically

Author: Suggestion implemented ✅

In the :guilabel:`Cyclic Counting` section, change the value in the :guilabel:`Inventory Frequency
(Days)` field from the default `0`, if necessary.

.. image:: use_locations/use-locations-cyclic-counting.png
   :align: center
   :alt: Cyclic Counting section of new location creation form.

When different than `0`, the inventory count dates for products stored at this location are
automatically set at the defined frequency.

In the :guilabel:`Logistics` section, in the :guilabel:`Removal Strategy` field, click the drop-down
menu and select the :doc:`removal strategy <../removal_strategies>` for how items should be removed
from this location.

.. _inventory/location-hierarchy:

Create location hierarchies
===========================

The *Parent Location* setting on the new location form allows for a location to exist within a
warehouse, or below another location.

Every location can serve as a parent location, and every parent location can have multiple locations
within it, allowing for the creation of a hierarchical structure.

.. example::
   Location hierarchy could be organized so that a small refrigerator is located on a specific
   shelf, which is located in the warehouse.

>    .. image:: use_locations/use-locations-hierarchy.png
>      :align: center
>      :alt: Location hierarchy of small refrigerator, on a shelf in a warehouse.

Reviewer 2: Small fix - indent the image attributes by one more space :)
```suggestion
   .. image:: use_locations/use-locations-hierarchy.png
      :align: center
      :alt: Location hierarchy of small refrigerator, on a shelf in a warehouse.
```

Author: Nice catch - suggestion implemented ✅
